### ISSUE TITLE
speccy fails on external (http) references

### ISSUE THREAD

**Issue by techieshark** [wework/speccy#79]
## Detailed description

Either this is a bug or I'm doing something wrong. I'm running `speccy lint` on a file which references a remotely-hosted file and speccy seems to be treating it like a local file and complaining that it doesn't exist.

## Context

I'm developing an API and if there's an error in an API I use, I want to pass the third party API's error to the client. So in my API, I want to just say "errors will be of the type described here:" and refer to a third party spec hosted remotely. (Given that this issue appears unreported, perhaps this is not a common use case? How are other people dealing with these situations?)

## Possible implementation

Unsure w/o diving into the code.

## Your environment

* Node Version: 8.11.1
* Operating system and version (e.g. Ubuntu 16.04, Windows 7): OS X
* Speccy version: 0.7.1

To reproduce:

1. Download the simple [petstore example spec](https://raw.githubusercontent.com/OAI/OpenAPI-Specification/OpenAPI.next/examples/v3.0/petstore.yaml).

2. Change the error ref to refer to an externally hosted file, for example:

```
--- petstore.yaml.orig	2018-05-30 13:16:26.000000000 -0700
+++ petstore.yaml	2018-05-30 13:16:33.000000000 -0700
@@ -52,7 +52,7 @@
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/Error"
+                $ref: "http://raw.githubusercontent.com/stripe/openapi/master/openapi/spec3.yaml"
   /pets/{petId}:
     get:
       summary: Info for a specific pet
```
(The actual ref in the above example would have `#components/schemas/error` appended but this simpler example fails too.)

3. run `npx speccy lint  --skip openapi-tags --skip operation-tags --skip info-contact petstore.yaml --verbose` 

(Note, skipping a few lint rules above just because the example doesn't include tags or contact details, but error occurs either way.)

You'd expect to have it declared valid, but this error is reported:
```
GET /project/node_modules/speccy/rules/default.json
Found 17 rules in default: parameter-description,parameter-name-regex,operation-operationId,operation-summary-or-description,operation-tags,path-keys-no-trailing-slash,server-trailing-slash,openapi-tags,openapi-tags-alphabetical,reference-no-other-properties,pathItem-summary-or-description,example-value-or-externalValue,reference-components-regex,no-script-tags-in-markdown,info-contact,license-apimatic-bug,no-eval-in-descriptions
GET petstore.yaml
GET http://raw.githubusercontent.com/stripe/openapi/master/openapi/spec3.yaml 
Error: Invalid $ref, pointing to a missing or inaccessable file: ENOENT: no such file or directory, open 'http://raw.githubusercontent.com/stripe/openapi/master/openapi/spec3.yaml'
    at readFileAsync.then.then.catch.err (/project/node_modules/speccy/lib/resolver.js:230:21)
    at <anonymous>
Invalid $ref, pointing to a missing or inaccessable file: ENOENT: no such file or directory, open 'http://raw.githubusercontent.com/stripe/openapi/master/openapi/spec3.yaml'
```

**Comment by philsturgeon** [comment]
Fixed in #80.

**Comment by philsturgeon** [comment]
This fix is now released as part of v0.7.2.
